### ISSUE TITLE
Request - strong naming of NuGet packages

### ISSUE THREAD

**Issue by rdewaele** [hashicorp/terraform-cdk#3127]
### Description

Would it be possible to provide strong named assemblies for the CDKTF libraries published on NuGet?

My current impression is that this would be according to Microsoft guidelines:
https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming

> You should strong name your open-source .NET libraries. Strong naming an assembly ensures the most people can use it, and strict assembly loading only affects .NET Framework.

Thank you for your consideration.

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by mutahhir** [comment]
@jsteinich do you have any insights / opinions on this? Primarily, are there any downsides for strong naming that we should be aware of?

**Comment by jsteinich** [comment]
I've never personally used them. I've also not really come across libraries actually doing it.
From https://learn.microsoft.com/en-us/dotnet/standard/assembly/strong-named:

> For .NET Core and .NET 5+, strong-named assemblies do not provide material benefits. The runtime never validates the strong-name signature, nor does it use the strong-name for assembly binding.

**Comment by rdewaele** [comment]
> I've never personally used them. I've also not really come across libraries actually doing it. From https://learn.microsoft.com/en-us/dotnet/standard/assembly/strong-named:
> 
> > For .NET Core and .NET 5+, strong-named assemblies do not provide material benefits. The runtime never validates the strong-name signature, nor does it use the strong-name for assembly binding.

I am also pleased that the newer runtimes no longer have a need for strong naming. However, .NET Framework applications are still very common, where strong naming still matters. This is why their recommendation to strongly name open-source libraries (I quoted above) includes the notion that this only affects .NET Framework.

For us, all applications we have developed for clients involving .NET Framework, also involve strongly named packages. So far we have come across one (not counting CDKTF) dependency that we needed that was not, which we had to disassemble, recompile and publish to a private repository to be able to use. Because this is anecdotal, I can not be sure how representative this is for the CDKTF audience, but it our personal experience.

Note: We also saw different approaches in how project tackle this issue. E.g. DotNetRDF is strongly named by default, but Scriban publishes two packages: Scriban and Scriban.Signed. I am not sure what the rationale is for the split.
